File: algorithms/293.Flip_Game/Flip_Game.cc
class Solution {
public:
    vector<string> generatePossibleNextMoves(string s) {
        vector<string> result;
        for(int i = 0; i < (int)s.size() - 1; i++) {
            if(s[i] == s[i+1] && s[i] == '+'){
                string t = s;
                t[i] = t[i+1] = '-';
                result.push_back(t);
            }
        }
        return result;
    }
};

// Flip Game
class Solution {
public:
  vector<string> generatePossibleNextMoves(string s) {
    vector<string> r;
    for (int i = 0; i+1 < s.size(); i++)
      if (s[i] == '+' && s[i+1] == '+') {
        r.push_back(s);
        r.back()[i] = '-';
        r.back()[i+1] = '-';
      }
    return r;
  }
};

File: algorithms/293.Flip_Game/Flip_Game.py
<<<<<<< HEAD
class Solution(object):
    def generatePossibleNextMoves(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        result = []
        if not s:
            return result
        for i in range(len(s)-1):
            if s[i] == '+' and s[i+1] == '+':
                result += s[:i]+'--'+s[i+2:],
        return result
=======
class Solution(object):
    def generatePossibleNextMoves(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        result = []
        if not s:
            return result
        for i in range(len(s)-1):
            if s[i] == '+' and s[i+1] == '+':
                result += s[:i]+'--'+s[i+2:],
        return result
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
